### PR TITLE
Pull-to-refresh optional for mouse events

### PR DESCRIPTION
**Pull request by BenjaVR**
With these changes one can disable the mouse events for the pull-to-refresh feature. This way you have the option to only enable this feature on touch devices.

### CODE REVIEW

**Comment by M0ns1gn0r** [comment]
Please merge, the current behavior is very irritating.

**Comment by ethaizone** [comment]
I think same too. This PR should be merge.

**Comment by ankeetmaini** [comment]
Can you please also change the prop table in the readme?

File: src/index.tsx [hunk 1]
```diff
@@ -97,9 +98,14 @@ export default class InfiniteScroll extends Component<Props, State> {
       this.el.addEventListener('touchmove', this.onMove);
       this.el.addEventListener('touchend', this.onEnd);
 
-      this.el.addEventListener('mousedown', this.onStart);
-      this.el.addEventListener('mousemove', this.onMove);
-      this.el.addEventListener('mouseup', this.onEnd);
+      if (
+        this.props.pullDownToRefreshOnlyOnTouchEvents !== undefined &&
+        !this.props.pullDownToRefreshOnlyOnTouchEvents
```
Comment (ankeetmaini) [inline src/index.tsx:103]: just truthy will work I guess

if (this.props.pullDownToRefreshOnlyOnTouchEvents) {
  return;
}
// add event listeners

**Comment by ankeetmaini** [comment]
Could you also please tell the issue which this fixes? On touch events this would work seamlessly I assume?

**Comment by BenjaVR** [comment]
> Could you also please tell the issue which this fixes? On touch events this would work seamlessly I assume?

These changes can disable the 'pull down to refresh' for mouse events. Motivation is that this feature is something that comes from the mobile (= touch) world and can be really inconvenient when used on a desktop.

Here is a modified CodeSandbox to demonstrate a problem: https://codesandbox.io/s/goofy-dream-cfyx4?file=/src/index.js
Try to select the text of one 'row', without refreshing the whole list.

There are other solutions that can be applied on the consumer-side, such as blocking mouse events when dragging over text, or setting the `pullToRefresh` boolean to false based on a mobile/desktop-device detection mechanism ... But I don't think all of these solutions are that 'easy' to apply or even work that good when you want to get things to just work.

> Can you please also change the prop table in the readme?

Done!

File: src/index.tsx (hunk 1 above)
Comment (BenjaVR) [reply src/index.tsx:103]: This will prevent the code after this if statement to run, right?

**Comment by BenjaVR** [comment]
@ankeetmaini does this have any chance to get merged, or should I close the PR?

**Pull request closed without merging**